crates/file-based-storage/src/lib.rs

```diff
@@ -137,3 +181,37 @@ impl relayer_amplifier_state::State for MemmapState {
         })
     }
 }
+
+impl SolanaListenerState for MemmapState {
```

Comment:
Not sure if this traits correctly belong to this crate, but it's true that `MemmapState` is already coupled with other crates types. Thus i do not see the reasons to not add this logic here.

Reply:
I agree, this is a good place to keep it.